packages/contentful--react-apps-toolkit/src/useFieldValue.tsx

```diff
@@ -0,0 +1,81 @@
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import { useSDK } from './useSDK';
+
+/**
+ * Options to specify which value of the current entry should be read and updated.
+ */
+export interface UseFieldValueOptions {
+  /**
+   * Id of the field to read and update.
+   * Can be omitted when app is rendered in field location. When omitted, the field the app is rendered in is used.
+   */
+  fieldId?: string;
+  /**
+   * Locale to read and update.
+   * When omitted, default locale is used.
+   */
+  locale?: string;
+}
+
+export type UseFieldValueReturnValue<Value = unknown> = [
+  value: Value,
+  setValue: (newValue: Value) => Promise<void>
+];
+
+/**
+ * Returns field value and a function to update it.
+ * Must be wrapped by SDKProvider.
+ * Can only be used when app is rendered in field, sidebar or entry editor location.
+ *
+ * @param {UseFieldValueOptions} options Options to specify which value of the current entry should be read and updated.
+ * @returns {UseFieldValueReturnValue} Field value, function to update it
+ */
+export function useFieldValue<Value = unknown>({
+  fieldId,
+  locale,
+}: UseFieldValueOptions = {}): UseFieldValueReturnValue<Value> {
+  const sdk = useSDK();
+  const entryFieldApi = useMemo(() => {
+    if (!('entry' in sdk)) {
+      throw new Error(
+        '`useFieldValue` can only be used in field, sidebar or entry editor location'
+      );
+    }
+
+    const fieldIdWithDefault = fieldId ?? ('field' in sdk ? sdk.field.id : undefined);
+
+    if (!fieldIdWithDefault) {
+      throw new Error(
+        'Missing `fieldId`. The `fieldId` can only be omitted when your app is renderd in field location.'
```

Comment:
I am mildly opposed to this part of design. I would prefer a `useFieldValue` and `useEntryFieldValue`/`useFieldValueById` hook which would handle the two different use cases that this hook currently does.

I'd prefer this because I think it's slightly simpler. I find overloading functions like this a bit confusing.

Reply:
Is the issue about the hook having a different behavior in different locations? Or that you have to pass in an object instead of string arguments?

Reply:
My object is that this one hook tries to handle two different use cases. I don't think it's a big issue, and a lot of our existing APIs are overloaded in this way, but I do think it's a bit confusing, and probably harder to maintain than just having two hooks.

Reply:
As a user of the package, I think it is good to only have one hook that you can import. Even when we split things up and export two hooks, people will probably import the wrong one.

I think I can improve the current implementation, so both use cases are not mixed that much. (extract an internal hook with a mandatory fieldId)